python/pytest-selfie/pytest_selfie/plugin.py

```diff
@@ -32,8 +32,23 @@
 
 
 class FSImplementation(FS):
-    def assert_failed(self, message: str, expected=None, actual=None) -> Exception:
-        raise Exception(message)
+    def assert_failed(self, message, expected=None, actual=None) -> Exception:
+        if expected is None and actual is None:
+            return AssertionError(message)
+
+        expected_str = self._nullable_to_string(expected, "(null)")
+        actual_str = self._nullable_to_string(actual, "(null)")
+
+        if not expected_str and not actual_str and (expected is None or actual is None):
+            return self._comparison_assertion(message, "(null)", "(null)")
```

Comment:
Not quite. We don't want to pass `"(null)", "(null)"`, we want to pass `"(null)"` as the default value if the thing is null

https://github.com/diffplug/selfie/blob/e6c59053316e3492b3552a5dcf54c9ba598ee7e3/jvm/selfie-runner-kotest/src/commonMain/kotlin/com/diffplug/selfie/kotest/FSOkio.kt#L51-L53